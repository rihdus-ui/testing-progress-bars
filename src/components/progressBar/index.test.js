/**
 * Created by t_mehes on 3/7/17.
 */

import React from 'react';
import ProgressBar from './index.js';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';


describe('Progressbar', () => {

  let progressBarComponent = null;

  beforeAll(() => {
    progressBarComponent = renderer.create((<ProgressBar/>))
  });

  it('component render', () => {
    let tree = progressBarComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('component render', () => {

    let testProgressBars = []
      , testData = [
        {value: 4, max: 30},
        {value: 0, max: 1},
      ]
    ;

    beforeAll(() => {
      testData.forEach((test, i) => {
        testProgressBars[i] = shallow(<ProgressBar value={test.value} max={test.max}/>);
      });
    });

    it('with props', () => {
      testData.forEach((test, i) => {

        const testProgressBar = testProgressBars[i]
          , progressText = `${test.value}/${test.max}` // 20/100
          , progressLineStyleTransform = `translateX(-${(test.max - test.value) * 100 / test.max}%)` // translateX(-80%)
        ;

        it('should render the text correctly', () => {
          const progressTextEl = testProgressBar.find('.bar-text');
          expect(progressTextEl.text()).toBe(progressText);
        });

        it('should render the the bar with correct value', () => {
          const lineSvg = testProgressBar.find('.path')
          ;
          const lineStyle = lineSvg.node.props.style;
          console.log(lineStyle.transform);
          expect(lineStyle).not.toBe(undefined);
          expect(lineStyle.transform)
            .toBe(progressLineStyleTransform);
        });
      })
    });

  })

});

