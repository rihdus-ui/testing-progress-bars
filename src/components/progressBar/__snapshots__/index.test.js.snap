// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progressbar component render 1`] = `
<div
  className="progressBar"
>
  <div
    className="bar-container"
  >
    <svg
      className="bar-svg"
      height="100%"
      preserveAspectRatio="none"
      viewBox="0 0 100 10"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        className="path"
        strokeDasharray="100"
        strokeDashoffset={0}
        strokeWidth="10"
        style={
          Object {
            "transform": "translateX(NaN%)",
          }
        }
        x1="0"
        x2="100%"
        y1="5"
        y2="5"
      />
    </svg>
  </div>
  <div
    className="bar-text"
  >
    <span />
    /
    <span />
  </div>
</div>
`;
